File: frontend/src/types/insight.ts
export type InsightSessionStatus = "gathering" | "summarizing" | "ready" | "failed";
export type InsightRole = "user" | "assistant";

export interface InsightMessage {
  id: string;
  role: InsightRole;
  content: string;
  referenced_job_ids: string[] | null;
  created_at: string;
}

export interface InsightSession {
  id: string;
  image_id: string;
  image_filename: string | null;
  source_job_id: string | null;
  status: InsightSessionStatus;
  model_used: string | null;
  error: { message?: string } | null;
  created_at: string;
  updated_at: string;
}

export interface InsightSessionDetail extends InsightSession {
  messages: InsightMessage[];
}

File: volatility3/app/schemas/insight.py
import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict

from ..models.insight import InsightRole, InsightSessionStatus


class InsightSessionCreate(BaseModel):
    image_id: uuid.UUID


class InsightMessageOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    role: InsightRole
    content: str
    referenced_job_ids: list[str] | None
    created_at: datetime


class InsightSessionOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    image_id: uuid.UUID
    image_filename: str | None
    source_job_id: uuid.UUID | None
    status: InsightSessionStatus
    model_used: str | None
    error: dict[str, Any] | None
    created_at: datetime
    updated_at: datetime


class InsightSessionDetailOut(InsightSessionOut):
    messages: list[InsightMessageOut]


class InsightAvailability(BaseModel):
    available: bool
    detail: str | None = None
